src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -577,44 +573,33 @@ namespace xclhwemhal2 {
       }
     }
 
-    pt::ptree xml_project;
-    std::string sXmlFile;
-    sXmlFile.assign(args.m_xmlfile, args.m_xmlFileSize);
-    std::stringstream xml_stream;
-    xml_stream << sXmlFile;
-    pt::read_xml(xml_stream, xml_project);
-
-    // iterate platforms
-    int count = 0;
-    for (auto& xml_platform : xml_project.get_child("project"))
-    {
-      if (xml_platform.first != "platform")
-        continue;
-      if (++count > 1)
-      {
-        //Give error and return from here
-      }
-    }
-
-    std::string fpgaDevice="";
+    //CR-1116870 Changes Start
 
-    // iterate devices
-    count = 0;
-    for (auto& xml_device : xml_project.get_child("project.platform"))
+    std::string projectName = xrt_core::xclbin::get_project_name(args.m_top);
+    std::string fpgaDevice = "";
+    if (!projectName.empty())
     {
-      if (xml_device.first != "device")
-        continue;
+      pt::ptree xml_project;
+      std::string sXmlFile;
+      sXmlFile.assign(args.m_xmlfile, args.m_xmlFileSize);
+      std::stringstream xml_stream;
+      xml_stream << sXmlFile;
+      pt::read_xml(xml_stream, xml_project);
 
-      fpgaDevice = xml_device.second.get<std::string>("<xmlattr>.fpgaDevice");
-
-      if (++count > 1)
+      // iterate devices
+      for (auto& xml_device : xml_project.get_child("project.platform"))
       {
-        //Give error and return from here
+        if (xml_device.first != "device")
+          continue;
+
+        fpgaDevice = xml_device.second.get<std::string>("<xmlattr>.fpgaDevice");
```

Comment:
I think fpgaDevice is used to determine if this is versal or not?  If so, that should be abstracted as opposed to doing string compares.   If the XSA name could be used instead (still needs abstractions for versal test), then you could get that from xrt::xclbin if this class object was available here.